base-system/usrroot/usr/lib/hos/utils.sh

```diff
@@ -34,24 +34,18 @@ safe_download(){
 		return 1 # Error
 	fi
 
-	if [ ! -z "$FILE_HASH_URL" ]; then
-		## If hash cannot be downloaded, then it's not possible to verify file
-		if ! wget --no-cache --tries=3 --timeout=10 "$FILE_HASH_URL" -O "$TMP_HASH_FILE"; then
-			echo "Cannot download the hash file $FILE_HASH_URL" 1>&2
-			rm -rf "$TMP"
-			return 1 # Error
-		fi
-
-		## If hash does not match, do not save file
-		pushd "$TMP"
-		if ! sha256sum --status --check "$TMP_HASH_FILE"; then
-			echo "Downloaded file does not match it's hash, aborting. $FILE_URL" 1>&2
-			popd
-			rm -rf "$TMP"
-			return 1 # Error
-		fi
+	pushd "$TMP"
+	## Write hash into hash file
+	echo "$FILE_HASH" > "$TMP_HASH_FILE"
```

Comment:
Why not to ask for the single hash and append the file name? 
Or do you see some advantage from using also the filename? (eg. more security when changing wallpaper?)

Reply:
Actually that's a good idea leaving the hash only, I didn't though on that possibility :p